pkg/util/log/eventpb/telemetry.proto

```diff
@@ -71,3 +71,31 @@ message CapturedIndexUsageStats {
   bool is_unique = 10 [(gogoproto.jsontag) = ",omitempty"];
   bool is_inverted = 11 [(gogoproto.jsontag) = ",omitempty"];
 }
+
+// CreateChangefeedQuery
+message CreateChangefeedQuery {
+  CommonEventDetails common = 1 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "", (gogoproto.embed) = true];
+  CommonJobEventDetails job = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "", (gogoproto.embed) = true];
+
+  // The number of tables listed in the query that the changefeed is to run on.
+  int32 num_tables = 3 [(gogoproto.jsontag) = ",omitempty"];
+
+  // The type of sink being emitted to (ex: kafka, nodelocal, webhook-https).
+  string sink_type = 4 [(gogoproto.jsontag) = ",omitempty", (gogoproto.moretags) = "redact:\"nonsensitive\""];
+
+  // The behavior of emitted resolved spans.
+  string resolved = 5 [(gogoproto.jsontag) = ",omitempty", (gogoproto.moretags) =  "redact:\"nonsensitive\""];
+
+  // The data format being emitted (ex: JSON, Avro).
+  string format = 6 [(gogoproto.jsontag) = ",omitempty", (gogoproto.moretags) =  "redact:\"nonsensitive\""];
+}
+
+// ChangefeedFailed
+message ChangefeedFailed {
```

Comment:
i'm questioning why we would want to have the separate message for falures.
Why not have failure_reason above in the main changefeed telemetry message?  Even when changefeed creation fails, it is possible that the other fields (e..g format, sink type, etc)  could be initialized.  So, for example, we could see particular sink types seem to fail more often.

Reply:
I still think my question stands: why is changefeed creation should be any different from changfeed error event?  Wouldn't we want to know for example, how many webhook sinks with more than 5 tables fail?
When changefeed fails after the job is created, we have all of the information that we include in createChangefeed event...

Also, I kinda think that we should not be adding many changefeed event types. For example, wouldn't we want to also emit events if somebody alters changefeed?
We could probably use CreateChangefeed (though this should be renamed) to ChangefeedEvent maybe)?

Reply:
What would be the benefit of sharing a `ChangefeedEvent` and I would assume adding another string property to distinguish the actual changefeed-specific event type, compared to having multiple events but putting more information in `CommonChangefeedEventDetails`?  

There's an idea for a Backfill Event with information like backfill_type, backfill_length, and backfill_size.  If we had multiple events like that with very specific information, I feel like a single `ChangefeedEvent` struct would get noisy.

Having just one struct would make parsing easier in the test, but I think it's worth that sacrifice for the clarity.

Reply:
> What would be the benefit of sharing a `ChangefeedEvent` and I would assume adding another string property to distinguish the actual changefeed-specific event type, compared to having multiple events but putting more information in `CommonChangefeedEventDetails`?
> 
> There's an idea for a Backfill Event with information like backfill_type, backfill_length, and backfill_size. If we had multiple events like that with very specific information, I feel like a single `ChangefeedEvent` struct would get noisy.
> 
> Having just one struct would make parsing easier in the test, but I think it's worth that sacrifice for the clarity.

Argh... I didn't realize this pr had LG.  That's unfortunate.  I have not for example, gave much thought to tagged errors, etc.  I think I made a mistake LG-ing instead of some other pr.

Thank you for moving to changefeed common details.  This addresses my biggest criticism.
I'm still not clear about the split into multiple message types -- there is "no prior" art for this (afaik other systems don't differentiate), nor do I know of any guidance for the telemetry usage.